drivers/rilmodem/sim.c

```diff
@@ -676,13 +676,20 @@ static void sim_send_set_uicc_subscription(struct ofono_sim *sim, int slot_id,
 {
 	struct sim_data *sd = ofono_sim_get_data(sim);
 	struct parcel rilp;
+	gint request_number;
 
 	DBG("");
 
+	if (g_ril_get_version(sd->ril) < 10 &&
+			sd->vendor == OFONO_RIL_VENDOR_QCOM_MSIM)
+		request_number = QCOM_MSIM_RIL_REQUEST_SET_UICC_SUBSCRIPTION;
+	else
+		request_number = RIL_REQUEST_SET_UICC_SUBSCRIPTION;
+
```

Comment:
Please return at the beginning of the function if version < 10 but this is not OFONO_RIL_VENDOR_QCOM_MSIM.